exchanges/deribit/deribit.go

```diff
@@ -0,0 +1,2773 @@
+func (d *Deribit) SubmitEdit(ctx context.Context, arg *OrderBuyAndSellParams) (*PrivateTradeData, error) {
+	if arg == nil {
+		return nil, fmt.Errorf("%w parameter is required", common.ErrNilPointer)
+	}
+	if arg.OrderID == "" {
+		return nil, fmt.Errorf("%w, order id is required", errInvalidID)
+	}
+	if arg.Amount <= 0 {
+		return nil, errInvalidAmount
+	}
+	params := url.Values{}
+	params.Set("order_id", arg.OrderID)
+	params.Set("amount", strconv.FormatFloat(arg.Amount, 'f', -1, 64))
+	if arg.PostOnly {
+		params.Set("post_only", "true")
+	}
+	if arg.RejectPostOnly {
+		params.Set("reject_post_only", "true")
+	}
+	if arg.ReduceOnly {
+		params.Set("reduce_only", "true")
+	}
+	if arg.MMP {
+		params.Set("mmp", "true")
+	}
+	if arg.TriggerPrice != 0 {
+		params.Set("trigger_price", strconv.FormatFloat(arg.TriggerPrice, 'f', -1, 64))
+	}
+	if arg.Advanced != "" {
+		params.Set("advanced", arg.Advanced)
+	}
+	if arg.Price > 0 {
+		params.Set("price", strconv.FormatFloat(arg.Price, 'f', -1, 64))
+	}
+	var resp *PrivateTradeData
+	return resp, d.SendHTTPAuthRequest(ctx, exchange.RestFutures, matchingEPL, http.MethodGet, submitEdit, params, &resp)
+}
+
+// EditOrderByLabel submits an edit order request sorted via label
+func (d *Deribit) EditOrderByLabel(ctx context.Context, arg *OrderBuyAndSellParams) (*PrivateTradeData, error) {
+	if arg == nil {
+		return nil, fmt.Errorf("%w parameter is required", common.ErrNilPointer)
+	}
+	params, err := checkInstrument(arg.Instrument)
+	if err != nil {
+		return nil, err
+	}
+	if arg.Amount <= 0 {
+		return nil, errInvalidAmount
+	}
+	if arg.Label != "" {
+		params.Set("label", arg.Label)
+	}
+	params.Set("amount", strconv.FormatFloat(arg.Amount, 'f', -1, 64))
+	if arg.PostOnly {
+		params.Set("post_only", "true")
+	}
+	if arg.RejectPostOnly {
+		params.Set("reject_post_only", "true")
+	}
+	if arg.ReduceOnly {
+		params.Set("reduce_only", "true")
+	}
+	if arg.MMP {
+		params.Set("mmp", "true")
+	}
+	if arg.TriggerPrice != 0 {
+		params.Set("trigger_price", strconv.FormatFloat(arg.TriggerPrice, 'f', -1, 64))
+	}
+	if arg.Advanced != "" {
+		params.Set("advanced", arg.Advanced)
+	}
+	var resp *PrivateTradeData
+	return resp, d.SendHTTPAuthRequest(ctx, exchange.RestFutures, nonMatchingEPL, http.MethodGet,
+		editByLabel, params, &resp)
+}
+
+// SubmitCancel sends a request to cancel the order via its orderID
+func (d *Deribit) SubmitCancel(ctx context.Context, orderID string) (*PrivateCancelData, error) {
+	if orderID == "" {
+		return nil, fmt.Errorf("%w, no order ID specified", errInvalidID)
+	}
+	params := url.Values{}
+	params.Set("order_id", orderID)
+	var resp *PrivateCancelData
+	return resp, d.SendHTTPAuthRequest(ctx, exchange.RestFutures, matchingEPL, http.MethodGet,
+		submitCancel, params, &resp)
+}
+
+// SubmitCancelAll sends a request to cancel all user orders in all currencies and instruments
+func (d *Deribit) SubmitCancelAll(ctx context.Context) (int64, error) {
```

Comment:
While it has been updated to respond with an int, it has been updated to also have a `detailed` bool, to return detailed results as highlighted here:
https://docs.deribit.com/#detailed-response-for-private-cancel_all-and-private-cancel_by_label-methods

So any function that has `cancel_all` within its path now can have a `detailed` param. aka:
- `SubmitCancelAllByCurrency`
- `SubmitCancelAllByKind`
- `SubmitCancelAllByInstrument`
- `SubmitCancelAll` 
- `SubmitCancelByLabel`

So when `detailed` is set to `true`, it will return with a different response, and we need to be able to handle both

Any usage of `cancel_all*` functions within our code like in `deribit_wrapper` should have `detailed` set to `true`